Add OpenVoice2 and MeloTTS

apinge opened the pull request:
This PR aims to add an enabling notebook for OpenVoice2 and MeloTTS.
Compared to OpenVoice v1, OpenVoice v2 uses MeloTTS as the BaseSpeaker and adds multi-lingual support.
Please ref  to the following pages:

* https://huggingface.co/myshell-ai/OpenVoiceV2
* https://github.com/myshell-ai/OpenVoice/blob/main/docs/USAGE.md#openvoice-v2. 


In this PR, I chose  [MeloTTS-English-v3](https://huggingface.co/myshell-ai/MeloTTS-English-v3) and [MeloTTS-Chinese](https://huggingface.co/myshell-ai/MeloTTS-Chinese) models as the BaseSpeaker.

**Tickets:**
[CVS-157888](https://jira.devtools.intel.com/browse/CVS-157888)

Commit:
add openvoice2

Commit:
remove redundant comments

Commit:
add audio play and update gradio

apinge commented:
@eaidova I have completed the model conversion and inference, but I encountered an issue with the Gradio interactive demo. Could you please kindly review and help?

sammysun0711 commented:
For Gradio demo issue w/o return value, it was not introduced by this PR. I found it can be reproduced with https://github.com/openvinotoolkit/openvino_notebooks/blob/latest/notebooks/openvoice/openvoice.ipynb, @eaidova, @yatarkan, could you please have a look for this issue. Thanks!
<img width="851" alt="Screenshot 2025-05-13 213153" src="https://github.com/user-attachments/assets/20fbed27-ba01-4e67-9dd3-60e8f2044741" />

Python env: 
[gradio_issue_env.txt](https://github.com/user-attachments/files/20188980/gradio_issue_env.txt)

Similar issue here: https://github.com/openvinotoolkit/openvino_notebooks/issues/2868
It cannot be solved by 
`python
import os
os.environ['no_proxy'] = "localhost,127.0.0.1" 
os.environ['NO_PROXY'] = "localhost,127.0.0.1" 
`
or 
`
export no_proxy="localhost,127.0.0.1"
export NO_PROXY="localhost,127.0.0.1"
`

Commit:
update gradio_helper.py

Commit:
add wordlist

eaidova commented:
<p>can this repo be installed with --no-deps for avoid such old dependencies pinning?</p>

---

 <a href='https://app.reviewnb.com/openvinotoolkit/openvino_notebooks/pull/2930/discussion/#comment-2876516310'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

Commit:
regulate the format, update download_from_hf_hub and clean unidict

Commit:
update all download checkpoint's path

Commit:
update notebook readme

Commit:
update the nltk, pip install options

apinge commented:
> can this repo be installed with --no-deps for avoid such old dependencies pinning?
> 
> [View entire conversation](https://app.reviewnb.com/openvinotoolkit/openvino_notebooks/pull/2930/discussion/#comment-2876516310) on ReviewNB

I included the `--no-deps` option during the pip installation.

apinge commented:
> View / edit / reply to [this conversation](https://app.reviewnb.com/openvinotoolkit/openvino_notebooks/pull/2930/discussion/#comment-2885869693) on ReviewNB
> 
> _eaidova commented on 2025-05-16T07:20:53Z_ ----------------------------------------------------------------
> 
> Line #27.    collect_telemetry("openvoice2.ipynb")
> please align with notebook file name

The ipynb file name has been changed.

apinge commented:
> View / edit / reply to [this conversation](https://app.reviewnb.com/openvinotoolkit/openvino_notebooks/pull/2930/discussion/#comment-2885869736) on ReviewNB
> 
> _eaidova commented on 2025-05-16T07:20:54Z_ ----------------------------------------------------------------
> 
> Line #76.    !python -m unidic download
> please clean up cell output

A cleanup command has been added in the last cell.

apinge commented:
> View / edit / reply to [this conversation](https://app.reviewnb.com/openvinotoolkit/openvino_notebooks/pull/2930/discussion/#comment-2885869788) on ReviewNB
> 
> _eaidova commented on 2025-05-16T07:20:55Z_ ----------------------------------------------------------------
> 
> Line #5.        local_path.mkdir(exist_ok=True)
> there is no need to create directory for downloading, as I know (hub will create it if it does not exists) you can use hf_hub_download directly without wrapping to function

This line has been removed.

apinge commented:
> import nltk
> 
> nltk.download('averaged_perceptron_tagger_eng')

The nltk download command has been added to the second cell, and the cleanup code has been included in the last cell.

apinge commented:
> View / edit / reply to [this conversation](https://app.reviewnb.com/openvinotoolkit/openvino_notebooks/pull/2930/discussion/#comment-2886362274) on ReviewNB
> 
> _aleksandr-mokrov commented on 2025-05-16T10:44:59Z_ ----------------------------------------------------------------
> 
> Line #15.        demo.queue(max_size=1).launch(debug=True, height=1000)
> Here is the error:
> 
> TypeError: predict_impl() got multiple values for argument 'speed'
> Looks like speed gets a values that passes for agree argument. Move it or pass all arguments by name.

I removed the speed arguments from the `predict_impl` and set the speed to a fixed value of 1.0.

eaidova commented:
> > View / edit / reply to [this conversation](https://app.reviewnb.com/openvinotoolkit/openvino_notebooks/pull/2930/discussion/#comment-2885869736) on ReviewNB
> > _eaidova commented on 2025-05-16T07:20:54Z_ ----------------------------------------------------------------
> > Line #76.    !python -m unidic download
> > please clean up cell output
> 
> A cleanup command has been added in the last cell.

@apinge  I mean different thing. Dependencies installation cell has large logs and ends by KeyBoardInterupt. It does not bring any useful information for user (only info about your testing environment) that is why we prefer to remove cell output logs from notebook for such requirements installation cells

Commit:
remove all cell logs

Commit:
remove demo close

Commit:
add nltk install

Commit:
clean wrong pxid

Commit:
reformat the file by black

Commit:
add scarf pixel

Commit:
remove japanese modules

apinge commented:
> The ipynb file name has been changed.

I've "Clean All Outputs" in the notebooks.

apinge commented:
> View / edit / reply to [this conversation](https://app.reviewnb.com/openvinotoolkit/openvino_notebooks/pull/2930/discussion/#comment-2889859421) on ReviewNB
> 
> _eaidova commented on 2025-05-19T07:06:51Z_ ----------------------------------------------------------------
> 
> Manual clsing is needed only for launching demo without debug mode, debug=True flag in gradio demo does not allow running cells in parallel (it blocks notebook execution( and you will need to close it by stopping execution. So for this situation, there is no need in this cell

This cell has been removed.

apinge commented:
> View / edit / reply to [this conversation](https://app.reviewnb.com/openvinotoolkit/openvino_notebooks/pull/2930/discussion/#comment-2876516310) on ReviewNB
> 
> _sammysun0711 commented on 2025-05-13T13:23:51Z_ ----------------------------------------------------------------
> 
> Line #52.        "transformers==4.27.4",
> **Edit: **
> 
> Why we need to limit the transformers for old version? It seems that all version limitation inherit from MeloTTS repo: [MeloTTS/requirements.txt at main · myshell-ai/MeloTTS](https://github.com/myshell-ai/MeloTTS/blob/main/requirements.txt)
> 
> _eaidova commented on 2025-05-16T07:17:31Z_ ----------------------------------------------------------------
> 
> can this repo be installed with --no-deps for avoid such old dependencies pinning?

I tried using some modules without the `--no-deps` option, but there are other necessary dependencies. Additionally, I removed some Japanese-related modules, not only because this notebook doesn't convert Japanese TTS models, but also because the Japanese-related modules introduce dependency issues on higher versions of Python on Windows (https://github.com/myshell-ai/MeloTTS/issues/126). For this reason, I also modified some code in the MeloTTS cloned repository (specifically in cell 3)."

Commit:
move the pip install of nncf to install dependenies

eaidova committed:
Merge branch 'latest' into openvoice2

aleksandr-mokrov commented on notebooks/openvoice2-and-melotts/openvoice2-and-melotts.ipynb:
```diff
@@ -0,0 +1,1263 @@
+{
```
<pre class="ql-syntax" spellcheck="false">Line #62.        if len(prompt) &gt; 50:
</pre><p>It calculates the number of characters. To calculate words, use len(prompt.split())</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/openvinotoolkit/openvino_notebooks/pull/2930/discussion/#comment-2097931439'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='43'/>

apinge replied:
I've updated the code.

Commit:
fix botocore and prompt.split()

Commit:
make gradio install with dependencies

Commit:
add encoding utf8

Commit:
ignore treon docker in ci for ffmpeg

Commit:
skip macos-13

Commit:
Merge branch 'latest' into openvoice2

sammysun0711 commented:
@eaidova, @aleksandr-mokrov, we fixed all review comments and all test case passed. Could you please kindly check if it is ready to merge?
